doc/languages-frameworks/rust.section.md

```diff
@@ -64,6 +65,14 @@ When the `Cargo.lock`, provided by upstream, is not in sync with the
 added in `cargoPatches` will also be prepended to the patches in `patches` at
 build-time.
 
+The `useRealVendorConfig` parameter tells cargo-vendor to include a Cargo
```

Comment:
Do we really have an incompatibility compared to the current state? Is the output different for packages that build at the moment? I would like to enable this by default otherwise.

Reply:
Otherwise a warning or should be printed otherwise people will never know about this option or bother to migrate.

Reply:
The incompatibility is: this changes the expected `cargoSha256`. So two approaches:
* we use the new scheme forcibly from now on and break every current hash
* or we deprecate the old behavior, let say one nixos release for people to switch to the new behavior, and then remove the old behavior.
This PR implements the second approach.
I agree about the warning.

Reply:
So you would not just add this:

 ```
  [source.crates-io]
  registry = 'https://github.com/rust-lang/crates.io-index'
  replace-with = 'vendored-sources'
  [source.vendored-sources]
  directory = '$(pwd)/$cargoDepsCopy'
```

to keep the checksum of existing packages in tact and then append the new content as required by the Cargo.toml?

Reply:
Or is there a problem with this approach?

Reply:
the right approach is to *remove* the config when it only contains crates.io. But this is a great idea indeed. I just pushed a commit which should not break any cargoSha256. It is a mass rebuild though. (it rebuilds cargo)